provider/pkg/provider/deployment_settings.go

```diff
@@ -529,7 +530,15 @@ func (ds *PulumiServiceDeploymentSettingsResource) Read(req *pulumirpc.ReadReque
 	}
 	settings, err := ds.client.GetDeploymentSettings(ctx, stack)
 	if err != nil {
+		statusCode := pulumiapi.GetErrorStatusCode(err)
+		if statusCode == http.StatusNotFound {
+			// deleteResponse causes the resource to be deleted from the state.
+			var deleteResponse = &pulumirpc.ReadResponse{Id: "", Properties: nil}
+			// If it's a 404 error, this resource was probably deleted.
+			return deleteResponse, nil
+		}
```

Comment:
This fixes the problem for DeploymentSettings, but it occurs to me that all of our other resources probably have the same bug. Can we do this check one layer up in the provider instead of the resource? _I think_ this would cover it:

https://github.com/pulumi/pulumi-pulumiservice/blob/main/provider/pkg/provider/provider.go#L205-L209

Aside: several resources don't implement `Read` (ugh) but we don't need to fix that here. It would be so nice to some day generate this provider from an OpenAPI spec!

Reply:
I will take a look tomorrow morning about that, thx 🙇

Reply:
I just found teams is handling the "not found" in the way I mention yesterday to you, returning nil instead of propagating the error, I will change to handle it in the same way https://github.com/pulumi/pulumi-pulumiservice/blob/main/provider/pkg/provider/team.go#L192-L194 (unless the api is returning a 404 too and it is just returning an error? https://github.com/pulumi/pulumi-pulumiservice/blob/master/provider/pkg/internal/pulumiapi/teams.go#L108-L112)

For webhooks, it is weird, I guess it is returning a 404 too? I will ping you to see how can I actually test it (I dont want to use pulumi org and my personal one does not have it enabled https://github.com/pulumi/pulumi-pulumiservice/blob/main/provider/pkg/provider/webhook.go#L374-L419)

Reply:
Confirmed, teams and webhooks suffer from the same thing
<img width="1028" alt="image" src="https://github.com/pulumi/pulumi-pulumiservice/assets/5647310/e25e0396-04d2-4a10-9d9d-043b66f763ac">

<img width="659" alt="image" src="https://github.com/pulumi/pulumi-pulumiservice/assets/5647310/ccfcb189-c8be-4d16-b129-73a3fb645077">

I will add those 2 tomorrow morning and ask you again for review, thx for bringing it up

Reply:
<img width="670" alt="image" src="https://github.com/pulumi/pulumi-pulumiservice/assets/5647310/ac4c8cd6-1b7a-4074-ba89-afa1f19cdc29">


<img width="669" alt="image" src="https://github.com/pulumi/pulumi-pulumiservice/assets/5647310/1177377a-1a21-44a0-865f-69e4b3989d45">